### PR TITLE
Install Bazel on CentOS

### PR DESCRIPTION
**Pull request by michelleirvine**
Is the `which` command the only difference between the requirements and steps for Ubuntu and CentOS? I've told users to follow the Ubuntu steps - is this okay?

Is our support for CentOS considered "experimental" or "beta"? Which versions of this OS do we currently test?

### CODE REVIEW

**Comment by damienmg** [comment]
So we test for CentOS 6.7 and Fedora 23 (basically because CentOS 6 is harder than 7 and needs to install a custom gcc).

The general installation instruction should be for Linux (any distribution) and we should try to list dependencies as much as possible then list the command list for the distrib we know how (ubuntu and centos). The docker file at https://github.com/bazelbuild/bazel/blob/master/src/test/docker/Dockerfile.centos6.7 give the command for centos

There is also freebsd that is now supported, cc @aehlig to give instruction but AFAICT this just require to install bazel from freebsd ports.

side-note: windows is no longer experimental in support, we should just remove experimental support from everywhere.

File: site/docs/install-ubuntu.md [hunk 1]
```diff
@@ -3,25 +3,28 @@ layout: documentation
...
+*   [Use the binary installer](#install-with-installer-linux)
 *   [Compile Bazel from source](install-compile-source.md)
 
 Bazel comes with two completion scripts. After installing Bazel, you can:
 
 *   access the [bash completion script](install.md)
 *   install the [zsh completion script](install.md)
 
-## <a name="install-on-ubuntu"></a> Using Bazel custom APT repository (recommended)
+## <a name="install-on-linux"></a> Using Bazel custom APT repository (recommended)
```
Comment (damienmg) [inline site/docs/install-ubuntu.md]: This should be ubuntu (APT repository are specific to debian's system, and we don't test them for debian, only for ubuntu).

File: site/docs/install.md [hunk 2]
```diff
@@ -7,9 +7,10 @@ title: Installing Bazel
 
 See the instructions for installing Bazel on:
 
-*   [Ubuntu Linux (16.04, 15.10, and 14.04)](install-ubuntu.md)
+*   [Linux](install-ubuntu.md)
 *   [Mac OS X](install-os-x.md)
 *   [Windows (experimental)](install-windows.md)
+*   [CentOS (experimental)](install-ubuntu.md)
```
Comment (damienmg) [inline site/docs/install.md]: remove line

File: site/docs/install.md [hunk 3]
```diff
@@ -7,9 +7,10 @@ title: Installing Bazel
 
 See the instructions for installing Bazel on:
 
-*   [Ubuntu Linux (16.04, 15.10, and 14.04)](install-ubuntu.md)
+*   [Linux](install-ubuntu.md)
 *   [Mac OS X](install-os-x.md)
 *   [Windows (experimental)](install-windows.md)
```
Comment (damienmg) [inline site/docs/install.md]: remove experimental

File: site/docs/install-ubuntu.md (hunk 1 above)
Comment (michelleirvine) [inline site/docs/install-ubuntu.md]: done.

File: site/docs/install.md (hunk 3 above)
Comment (michelleirvine) [inline site/docs/install.md]: done

File: site/docs/install.md (hunk 2 above)
Comment (michelleirvine) [inline site/docs/install.md]: done

File: site/docs/install.md [hunk 4]
```diff
@@ -9,8 +9,7 @@ See the instructions for installing Bazel on:
 
 *   [Linux](install-ubuntu.md)
 *   [Mac OS X](install-os-x.md)
-*   [Windows (experimental)](install-windows.md)
-*   [CentOS (experimental)](install-ubuntu.md)
+*   [Windows](install-windows.md)
 
 For other platforms, you can try to [compile from source](install-compile-source.md).
```
Comment (michelleirvine) [inline site/docs/install.md]: Are there platforms other than Linux, macOS, or Windows? If not, I'll remove this line (we point to compile from source from the other topics)

File: site/docs/install-ubuntu.md [hunk 5]
```diff
@@ -3,25 +3,29 @@ layout: documentation
 title: Installing Bazel on Ubuntu
 ---
 
-# <a name="ubuntu"></a>Install Bazel on Ubuntu
+# <a name="ubuntu"></a>Install Bazel on Linux
 
-Supported Ubuntu Linux platforms:
+Supported Linux platforms:
```
Comment (michelleirvine) [inline site/docs/install-ubuntu.md]: What's the full list of supported Linux platforms? Should I list everything that has an installer? (see https://github.com/bazelbuild/bazel/releases)

File: site/docs/install-ubuntu.md [hunk 6]
```diff
@@ -3,25 +3,29 @@ layout: documentation
...
+   *   14.04 (LTS)
+*  CentOS 6.7
 
-Install Bazel on Ubuntu using one of the following methods:
+Install Bazel on Linux using one of the following methods:
 
-*   [Use our custom APT repository (recommended)](#install-on-ubuntu)
-*   [Use the binary installer](#install-with-installer-ubuntu)
+*   [Use our custom APT repository (Ubuntu only)](#install-on-ubuntu)
+*   [Use the binary installer](#install-with-installer-linux)
```
Comment (michelleirvine) [inline site/docs/install-ubuntu.md]: I don't see an installer for CentOS -- can you only compile from source to install on that OS?

File: site/docs/install.md [hunk 7]
```diff
@@ -7,9 +7,9 @@ title: Installing Bazel
 
 See the instructions for installing Bazel on:
 
-*   [Ubuntu Linux (16.04, 15.10, and 14.04)](install-ubuntu.md)
-*   [Mac OS X](install-os-x.md)
-*   [Windows (experimental)](install-windows.md)
+*   [Linux](install-ubuntu.md)
```
Comment (damienmg) [inline site/docs/install.md]: Ok I have read a bit more from the website, we actually should keep ubuntu linux here (with the version), revert the change on the ubuntu page, then maybe add the line "For other platforms, ..." something like (supported platform are CentOS 6.x and 7.x and Fedora 22).

Comment (michelleirvine) [inline site/docs/install.md]: What's the full list of supported platforms and which ones can you only install by compiling from source?

File: site/docs/install-ubuntu.md (hunk 5 above)
Comment (damienmg) [inline site/docs/install-ubuntu.md]: We support Windows 7+, macOS 10.11, Ubuntu Linux (14.04, 16.04 and 16.10 IIRC) with binary packages. Then we test compilation from source for debian stretch, centos 6.x,
 fedora 23 and ubuntu 14.04/16.04 (see https://github.com/bazelbuild/bazel/tree/master/src/test/docker) and also for FreeBSD 11 and 12. We also maintain freebsd port of Bazel (@aehlig).

File: site/docs/install-ubuntu.md (hunk 6 above)
Comment (damienmg) [inline site/docs/install-ubuntu.md]: For now yes, I believe there is a repository maintained by someone at redhat that has a rpm now.

**Comment by damienmg** [comment]
Is this PR still up?

**Comment by buchgr** [comment]
FYI https://github.com/bazelbuild/bazel/pull/3721
